imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/TIFFImageReader.java

```diff
@@ -701,6 +661,18 @@ else if (bitsPerSample <= 0 || bitsPerSample > 16) {
         }
     }
 
+    private ImageTypeSpecifier create(int planarConfiguration, ColorSpace cs, int dataType, int significantSamples, int samplesPerPixel, boolean alpha, boolean alphaPremultiplied) {
```

Comment:
I kind of like this generalization. But the method should probably be named `createImageTypeSpecifier`.
Also, your change breaks the existing logic in several places where the original code would fall through and throw exception (ie. if someone introduced a new value for the `PlanarConfiguration` field), your new method will create a banded spec, which is incorrect, and will probably fail at some random place later.
I would prefer if you kept the original swich logic in this method, and added a `default` that simply throws an exception about the unsupported planar config.